agent/config.go

```diff
@@ -58,31 +131,95 @@ var defaultID = func() string {
 }()
 
 // Initialize the agent configuration from the provided .yaml file
-func InitConfig(path *string, config *koanf.Koanf) {
+func InitConfig(path *string) {
+	lock.Lock()
+	defer lock.Unlock()
+
 	config.Load(defaultConfig, nil)
 	log.Infof("Init config from file: %s", *path)
 	if err := config.Load(file.Provider(*path), yaml.Parser()); err != nil {
 		log.Errorf("Error loading config: %v", err)
 	}
-	validate(config)
+	validate()
 	log.Debugf(config.Sprint())
 }
 
+// Parse topic configuration
+func parseTopics(topics []string, direction Direction) []Topic {
+	var all []Topic
+	for _, t := range topics {
+		s := strings.Split(t, ":")
+		if len(s) == 1 {
+			all = append(all, Topic{
+				sourceName:      strings.TrimSpace(s[0]),
+				destinationName: strings.TrimSpace(s[0]),
+				direction:       direction,
+			})
+		} else if len(s) == 2 {
+			// Push from source topic to destination topic
+			var src = strings.TrimSpace(s[0])
+			var dst = strings.TrimSpace(s[1])
```

Comment:
Nit: Should we just put this in the else statement for clarity?

Reply:
This is intentional to initialise the variables, otherwise I'd have to do something horrible like...
```go
var src, dst = ""
if direction == Push {
    ...
}
```